crates/pep440-rs/src/tracked_from_str.rs

```diff
@@ -0,0 +1,14 @@
+use std::path::Path;
+
+/// Like FromStr, but with information about where the str came from
+pub trait TrackedFromStr: Sized {
```

Comment:
I am not entirely sure this is the right package, but AFAIK, it's the only crate that everything that needs it can depend on....

Reply:
I want to experiment with a slightly different approach here. Instead of a trait, could we just always use `None` for the source, and then add a method to set the source _after_ parsing? Since the parsing itself doesn't depend on the source at all, right?

Reply:
(I can give it a try.)

Reply:
Sure. If you can make it work, all good!